forte/data/data_store.py

```diff
@@ -937,7 +940,46 @@ def _parse_onto_file(self):
         """
         if self._onto_file_path is None:
             return
-        raise NotImplementedError
+
+        merged_scheme, merged_dict = [], []
+        DataStore.onto_gen.parse_ontology_spec(
+            self._onto_file_path, merged_scheme, merged_dict
+        )
+
+        for onto in merged_scheme:
+            entry_name = onto["entry_name"]
+            if entry_name in DataStore._type_attributes:
+                continue
+
+            attr_dict = {}
+            idx = constants.ATTR_BEGIN_INDEX
+            if "attributes" in onto:
+                for d in onto["attributes"]:
+                    name = d["name"]
+                    attr_dict[name] = idx
+                    idx += 1
+            entry_dict = {}
+            entry_dict["parent_class"] = set()
+            entry_dict["parent_class"].add(onto["parent_entry"])
+            entry_dict["attributes"] = attr_dict
+            DataStore._type_attributes[entry_name] = entry_dict
```

Comment:
Are you trying to parse the parent information yourself?

1. You should be using existing functions
    - https://github.com/asyml/forte/blob/master/forte/data/ontology/ontology_code_generator.py#L680
    - Here is an example usage: https://github.com/asyml/forte/blob/6405c96f179faef3dd58d9d5232962675fceb652/forte/pipeline.py#L672
2. Do your code here properly traverse the tree to make sure we have all the ancestor information available in the ontology file? 

Also notice the EntryTree object actually provide all the attributes: https://github.com/asyml/forte/blob/master/forte/data/ontology/code_generation_objects.py#L798

Also, note that the attributes are in `set` (unordered), remember to sort them.

Reply:
I used parse_ontology_spec in OntologyCodeGenerator, but the returned merged_scheme does not have the same structure as used in DataStore. This piece of code is trying to convert them to store in `_type_attributes`.